### PR TITLE
[DataGrid] Remove GridFormatterParams completely

### LINKED ISSUE #12658: The GridValueFormatterParams interface has NOT been removed in v7
### Steps to reproduce

According to the [migration guide](https://mui.com/x/migration/migration-data-grid-v6) the `GridValueFormatterParams` interface should have been removed but it seems as if it still available and the following type of code does not raise any errors.

```typescript
import type {GridColDef, GridValueFormatterParams} from '@mui/x-data-grid-pro';

export const getColumnTimestamp = (field: string, options?: Partial<GridColDef>): GridColDef => ({
	field,
	resizable: true,
	headerName: 'Timestamp',
	width: 200,
	type: 'dateTime',
	valueFormatter: (params: GridValueFormatterParams<Date | null>): string =>
		params.value === null || !dateIsValid(params.value) ? '' : dateAsTimestamp(params.value),
	...(options ? options : {}),
});
```

### Current behavior

It is still possible in v7 possible to use the `GridValueFormatterParams` interface and therefore impossible to find the code parts that should be migrated.

### Expected behavior

The `GridValueFormatterParams` interface should no longer be available as specified in the migration guide

### Context

_No response_

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
System:
    OS: macOS 14.4.1
  Binaries:
    Node: 20.12.0 - ~/.nvm/versions/node/v20.12.0/bin/node
    npm: 10.5.1 - ~/.nvm/versions/node/v20.12.0/bin/npm
    pnpm: 8.15.6 - ~/.nvm/versions/node/v20.12.0/bin/pnpm
  Browsers:
    Chrome: 123.0.6312.87
    Edge: Not Found
    Safari: 17.4.1
  npmPackages:
    @emotion/react: 11.11.4 => 11.11.4
    @emotion/styled: 11.11.5 => 11.11.5
    @mui/core-downloads-tracker:  5.15.14
    @mui/icons-material: 5.15.14 => 5.15.14
    @mui/material: 5.15.14 => 5.15.14
    @mui/private-theming:  5.15.14
    @mui/styled-engine:  5.15.14
    @mui/system:  5.15.14
    @mui/types:  7.2.14
    @mui/utils:  5.15.14
    @mui/x-data-grid:  7.1.0
    @mui/x-license:  7.0.0
    @types/react: 18.2.74 => 18.2.74
    react: 18.2.0 => 18.2.0
    react-dom: 18.2.0 => 18.2.0
    typescript: 5.4.3 => 5.4.3
```
</details>

**Search keywords**: GridValueFormatterParams

### PR DESCRIPTION
**Pull request by romgrk**
Closes #12658 

We didn't complete this breaking change. This is now a bug at runtime for users, so we need to remove it even if we didn't do it at v7 release.

### CODE REVIEW

File: packages/x-data-grid-premium/src/tests/aggregation.DataGridPremium.test.tsx [hunk 1]
```diff
@@ -641,7 +641,7 @@ describe('<DataGridPremium /> - Aggregation', () => {
     it('should use the aggregation function valueFormatter if defined', () => {
       const customAggregationFunction: GridAggregationFunction = {
         apply: () => 'Agg value',
-        valueFormatter: (params) => `+ ${params.value}`,
+        valueFormatter: (value) => `+ ${value}`,
```
Comment (MBilalShafi) [inline packages/x-data-grid-premium/src/tests/aggregation.DataGridPremium.test.tsx:644]: Probably document this in the migration guide and release notes

File: docs/pages/x/api/data-grid/grid-aggregation-function.json
Comment (MBilalShafi) [inline docs/pages/x/api/data-grid/grid-aggregation-function.json:1]: The failing argos SS could be fixed by:
```diff
diff --git a/docs/data/data-grid/aggregation/AggregationValueFormatter.tsx b/docs/data/data-grid/aggregation/AggregationValueFormatter.tsx
index 1a3f4c5a1..e5be77639 100644
--- a/docs/data/data-grid/aggregation/AggregationValueFormatter.tsx
+++ b/docs/data/data-grid/aggregation/AggregationValueFormatter.tsx
@@ -48,7 +48,7 @@ const firstAlphabeticalAggregation: GridAggregationFunction<string, string | nul
       return sortedValue[0];
     },
     label: 'first alphabetical',
-    valueFormatter: (params) => `Agg: ${params.value}`,
+    valueFormatter: (value) => `Agg: ${value}`,
     columnTypes: ['string'],
   };
```

**Review by MBilalShafi — APPROVED** [review]
Interesting use case for the inevitable BCs during the stable phase.

P.S. Could deprecation be used instead?

File: packages/x-data-grid-premium/src/tests/aggregation.DataGridPremium.test.tsx (hunk 1 above)
Comment (romgrk) [reply packages/x-data-grid-premium/src/tests/aggregation.DataGridPremium.test.tsx:644]: It's already in. This was missed during the refactor.

**Comment by cherniavskii** [comment]
Thanks! I overlooked this in https://github.com/mui/mui-x/pull/11573

**Pull request merged**